dask/dataframe/indexing.py

```diff
@@ -41,30 +42,247 @@ def _meta_indexer(self):
         return self.obj._meta.iloc
 
     def __getitem__(self, key):
+        if isinstance(key, int):
+            key = slice(key, key+1)
+
+        if isinstance(key, (slice, list, np.ndarray, Index, Series)):
+            key = tuple([key])
 
-        # dataframe
-        msg = (
-            "'DataFrame.iloc' only supports selecting columns. "
-            "It must be used like 'df.iloc[:, column_indexer]'."
-        )
         if not isinstance(key, tuple):
-            raise NotImplementedError(msg)
+            raise ValueError("Expected slice or tuple or Dask Index, got %s" % str(key))
```

Comment:
@ryan-williams does eliminating the NotImplementedError here interfere with python figuring out which class handles an operation here?